bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/profile/ProfileTypeResource.java

```diff
@@ -128,13 +126,7 @@ private Predicate<ProfileType> matchesChannelUID(@Nullable String channelTypeUID
             // requested to filter against an unknown channel type -> do not return a ProfileType
             return t -> false;
         }
-        switch (channelType.getKind()) {
-            case STATE:
-                return t -> stateProfileMatchesProfileType(t, channelType);
-            case TRIGGER:
-                return t -> triggerProfileMatchesProfileType(t, channelType);
-        }
-        return t -> false;
+        return t -> profileTypeMatchesChannelType(t, channelType);
```

Comment:
Well, to me, now that the methods have been moved to the `ProfileType` ancestor interface, there is no need to test because any future profile type will have to implement them.
Did I misunderstand something here?

Reply:
There is a difference between TRIGGER and STATE channels, they serve different purposes. So for a TRIGGER channel type only TRIGGER profiles should be returned.

Reply:
Ah, yes, I see what you mean, it makes sense indeed.